scripts/SHI/scpSHI_Jetz_complete.R

```diff
@@ -0,0 +1,396 @@
+}
+)
+
+print("========== Map of suitable area generated ==========")
+
+#2.2 Load habitat preferences---------------------------------------------
+# df_IUCN_habitat_cat <- rl_habitats(sp,key = token)$result
+# 
+# #Load table with land cover equivalences need to be updated with Jung et al
+# df_IUCN_to_LC_categories <- read.csv(file.path(path_script,"SHI","IUCN_to_LC_categories.csv"),colClasses = "character") # PENDING PUT 0.5 TO MARGINAL HABITATS
+# df_IUCN_habitat_LC_cat <- left_join(df_IUCN_habitat_cat,df_IUCN_to_LC_categories, by="code")
+# LC_codes <- as.numeric(unique(df_IUCN_habitat_LC_cat$ESA_cod))
+
+#2.3 Hydrological features------------------------------------------------------
+
+
+#-------------------------------------------------------------------------------------------------------------------
+#3. Land Cover Values
+#-------------------------------------------------------------------------------------------------------------------
+#3.1 GFW data-------------------------------------------------------------------
+#forest base map
+cube_GFW_TC <-
+  load_cube(stac_path = "https://io.biodiversite-quebec.ca/stac/",
+            limit = 1000,
+            collections = c("gfw-treecover2000"),
+            bbox = sf_ext_crs,
+            srs.cube = srs_cube,
+            spatial.res = spat_res,
+            temporal.res = "P1Y",
+            t0 = "2000-01-01",
+            t1 = "2000-12-31",
+            resampling = "bilinear")
+
+cube_GFW_TC_range <- cube_GFW_TC |>
+  gdalcubes::filter_pixel(paste0("data >=", forest_threshold)) |>#filter to forest threshold 0-100
+  gdalcubes::apply_pixel("data/data") # turn into a forest presence map
+
+r_GFW_TC_range <- cube_to_raster(cube_GFW_TC_range , format="terra") # convert to raster format
+
+r_suitability_map_rescaled <- terra::resample(r_suitability_map,r_GFW_TC_range,method="mode") #Adjust scale of suitability map
+r_GFW_TC_range_mask <- r_GFW_TC_range |>
+  terra::classify(rcl=cbind(NA,0)) |> # turn NA to 0
+  terra::mask(r_suitability_map_rescaled) # crop to suitability map
+
+print("========== Base forest layer downloaded ==========")
+
+# Download forest loss maps and create different layers for each year to remove from forest
+s_obj <- stac("https://io.biodiversite-quebec.ca/stac/")
+
+it_obj <- s_obj |>
+  stac_search(collections = "gfw-lossyear",
+              bbox = sf_ext) |>
+  get_request()
+
+st <- gdalcubes::stac_image_collection(it_obj$features, asset_names = c("data")) # create image collection
+
+v <- cube_view(srs = srs_cube, extent = list(t0 = "2000-01-01", t1 = "2000-12-31", # create cube according to area of interest
+                                             left = sf_ext_crs['xmin'], right = sf_ext_crs['xmax'],
+                                             top = sf_ext_crs['ymax'], bottom =  sf_ext_crs['ymin']),
+               dx=spat_res, dy=spat_res, dt="P1Y",
+               resampling = "mode", aggregation = "mode") # TO CHANGE to proportions
```

Comment:
Is this load_prop_values() function  different form that we have in stac_functions.r?

Reply:
Yes, here I get all the values less or equal to the respective year, to have layers with the accumulated changes, but no the proportions.
This because the layer gfw-lossyear is one layer with the number of the year where there was a loss, so I need to have layers by time step with the accumulated loss.